test/test_nestedtensor.py

```diff
@@ -890,6 +888,21 @@ def grad_test_func(a, b, c, weight, bias=None):
         data = (a, b, c, weight)
         assert torch.autograd.gradcheck(grad_test_func, inputs=data)
 
+    def test_nested_tensor_softmax(self):
+        a = torch.randn(1, 2, requires_grad=True, dtype=torch.float64)
+        b = torch.randn(2, 2, requires_grad=True, dtype=torch.float64)
+        c = torch.randn(3, 2, requires_grad=True, dtype=torch.float64)
+
+        def grad_test_func(a, b, c, dim):
+            nt = torch.nested_tensor([a, b, c])
+            # This implicitly tests to_padded_tensor grads
+            d = torch.functional.F.softmax(nt, dim=dim)
+            return d.to_padded_tensor(0)
+
+        # softmax over last dim
+        data = (a, b, c, -1)
+        assert torch.autograd.gradcheck(grad_test_func, inputs=data)
```

Comment:
nit: use `gradcheck` from `torch.testing._internal.common_utils` which is basically just a wrapper around gradcheck that sets some defaults that we use in the test suite.

Reply:
Just tried this and that one internally calls torch.stack() which is not a registered op for nested tensors